pkgs/development/python-modules/tree-sitter-grammars/default.nix

```diff
@@ -0,0 +1,128 @@
+{ lib
+, buildPythonPackage
+, pytestCheckHook
+, tree-sitter
+, symlinkJoin
+, writeTextDir
+  # `name`: grammar derivation pname in the format of `tree-sitter-<lang>`
+, name
+, grammarDrv
+}:
+let
+  inherit (grammarDrv) version;
+
+  snakeCaseName = lib.replaceStrings [ "-" ] [ "_" ] name;
+  drvPrefix = "python-${name}";
+  langIdentOverrides = {
+    tree_sitter_org_nvim = "tree_sitter_org";
+  };
+  langIdent = langIdentOverrides.${snakeCaseName} or snakeCaseName;
+in
+assert (lib.assertMsg (!(grammarDrv.meta ? license)) ''
+  As of this writing, ${grammarDrv.pname} surprisingly doesn't have a license.
+  This trap is set here to guarantee that if it ever does have a license, this package will inherit the license.
+'');
+buildPythonPackage {
+  inherit version;
+  pname = drvPrefix;
+
+  src = symlinkJoin {
+    name = "${drvPrefix}-source";
+    paths = [
+      (writeTextDir "${snakeCaseName}/__init__.py"
+        ''
+          from ._binding import language
+
+          __all__ = ["language"]
+        ''
+      )
+      (writeTextDir "${snakeCaseName}/binding.c"
+        ''
+          #include <Python.h>
+
+          typedef struct TSLanguage TSLanguage;
+
+          TSLanguage *${langIdent}(void);
+
+          static PyObject* _binding_language(PyObject *self, PyObject *args) {
+              return PyLong_FromVoidPtr(${langIdent}());
+          }
+
+          static PyMethodDef methods[] = {
+              {"language", _binding_language, METH_NOARGS,
+              "Get the tree-sitter language for this grammar."},
+              {NULL, NULL, 0, NULL}
+          };
+
+          static struct PyModuleDef module = {
+              .m_base = PyModuleDef_HEAD_INIT,
+              .m_name = "_binding",
+              .m_doc = NULL,
+              .m_size = -1,
+              .m_methods = methods
+          };
+
+          PyMODINIT_FUNC PyInit__binding(void) {
+              return PyModule_Create(&module);
+          }
+        ''
+      )
+      (writeTextDir "setup.py"
+        ''
+          from platform import system
+          from setuptools import Extension, setup
+
+
+          setup(
+            name="${snakeCaseName}",
+            version="${version}",
+            packages=["${snakeCaseName}"],
+            ext_package="${snakeCaseName}",
+            ext_modules=[
+              Extension(
+                name="_binding",
+                sources=["${snakeCaseName}/binding.c"],
+                extra_objects = ["${grammarDrv}/parser"],
+                extra_compile_args=(
+                  ["-std=c11"] if system() != 'Windows' else []
+                ),
+                define_macros=[
+                  ("Py_LIMITED_API", "0x03080000"),
```

Comment:
What does the magic number/address `0x03080000` here mean, could you please add a comment?

Reply:
`0x03080000` defines the lowest Python version our extension supports to be 3.8

Official docs: <https://docs.python.org/3/c-api/stable.html#c.Py_LIMITED_API>

The usage here follows upstream examples:
- https://github.com/tree-sitter/tree-sitter-ocaml/blob/0b12614ded3ec7ed7ab7933a9ba4f695ba4c342e/setup.py#L49
- https://github.com/tree-sitter/tree-sitter-typescript/blob/4f3eb6655a1cd1a1f87ef10201f8e22886dcd76e/setup.py#L49
- https://github.com/tree-sitter/tree-sitter-ruby/blob/0ffe457fb6aabf064f173fd30ea356845cef2513/setup.py#L48

Reply:
That should be documented, and based on that you should also set `disabled = pythonOlder "3.8"`